services/m3dbnode/server/server.go

```diff
@@ -174,13 +174,24 @@ func Run(runOpts RunOptions) {
 		SetRetentionPeriod(cfg.CommitLog.RetentionPeriod).
 		SetBlockSize(cfg.CommitLog.BlockSize))
 
-	runtimeOptsMgr := m3dbruntime.NewOptionsManager(m3dbruntime.NewOptions().
+	runtimeOpts := m3dbruntime.NewOptions().
 		SetPersistRateLimitOptions(ratelimit.NewOptions().
 			SetLimitEnabled(true).
 			SetLimitMbps(cfg.Filesystem.ThroughputLimitMbps).
 			SetLimitCheckEvery(cfg.Filesystem.ThroughputCheckEvery)).
 		SetWriteNewSeriesAsync(cfg.WriteNewSeriesAsync).
-		SetWriteNewSeriesBackoffDuration(cfg.WriteNewSeriesBackoffDuration))
+		SetWriteNewSeriesBackoffDuration(cfg.WriteNewSeriesBackoffDuration)
+	if value := cfg.Tick.SeriesBatchSize; value > 0 {
```

Comment:
Yeah, I mean I can make `Tick TickConfiguration` be a pointer type `Tick *TickConfiguration` and hence nil-able field on `Configuration` perhaps - and as such able to easily to specify to use the default configuration instead of what is listed.

That might help avoid what you're calling out here.